Nix 0.22 merged the "Error" and "Errno" types and deprecated the conversion
operations between them. Nix 0.26 removed the deprecated stuff. The code
changes to deal with this were taken from the upstream commit described below,
the precise dependency changes are local to Debian.

commit 0e6b1a8651e7ff6a3c709b35587b7020b21c9f54
Author: James O. D. Hunt <james.o.hunt@intel.com>
Date:   Thu Nov 25 14:20:46 2021 +0000

    cargo: Update nix crate
    
    Updated the `nix` crate to the latest version as this allows `errno`
    handling to be simplified.
    
    Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,7 +44,7 @@
 version = "0.4"
 
 [dependencies.nix]
-version = "0.20.2"
+version = "0.26"
 
 [dependencies.protobuf]
 version = "2.0"
--- a/src/sync/channel.rs
+++ b/src/sync/channel.rs
@@ -22,8 +22,8 @@ use crate::ttrpc::Code;
 use crate::MessageHeader;
 
 fn retryable(e: nix::Error) -> bool {
-    use ::nix::{errno::Errno, Error};
-    e == Error::from_errno(Errno::EINTR) || e == Error::from_errno(Errno::EAGAIN)
+    use ::nix::Error;
+    e == Error::EINTR || e == Error::EAGAIN
 }
 
 fn read_count(fd: RawFd, count: usize) -> Result<Vec<u8>> {
